src/types/observablearray.ts

```diff
@@ -124,8 +133,14 @@ class ObservableArrayAdministration<T> implements IInterceptable<IArrayWillChang
 		if (oldLength !== this.lastKnownLength)
 			throw new Error("[mobx] Modification exception: the internal structure of an observable array was changed. Did you use peek() to change it?");
 		this.lastKnownLength += delta;
-		if (delta > 0 && oldLength + delta > OBSERVABLE_ARRAY_BUFFER_SIZE)
-			reserveArrayBuffer(oldLength + delta);
+		if (delta > 0 && oldLength + delta + 1 > OBSERVABLE_ARRAY_BUFFER_SIZE)
+			reserveArrayBuffer(oldLength + delta + 1);
+		if (delta > 0)
```

Comment:
I did a quick experiment, but without the other entries it didn't seem to
work on my IPad (only define the own prop on index 0), so I just released
it with the above setup, except that I wrapped the iOS bug detecting var
around it, to not slow down the other environments in the same way.

Op vr 8 jul. 2016 om 09:22 schreef Andy Kogut <notifications@github.com>:

> In src/types/observablearray.ts
> <https://github.com/mobxjs/mobx/pull/396#discussion_r70035120>:
>
> > @@ -124,8 +133,14 @@ class ObservableArrayAdministration<T> implements IInterceptable<IArrayWillChang
> >  		if (oldLength !== this.lastKnownLength)
> >  			throw new Error("[mobx] Modification exception: the internal structure of an observable array was changed. Did you use peek() to change it?");
> >  		this.lastKnownLength += delta;
> > -		if (delta > 0 && oldLength + delta > OBSERVABLE_ARRAY_BUFFER_SIZE)
> > -			reserveArrayBuffer(oldLength + delta);
> > +		if (delta > 0 && oldLength + delta + 1 > OBSERVABLE_ARRAY_BUFFER_SIZE)
> > +			reserveArrayBuffer(oldLength + delta + 1);
> > +		if (delta > 0)
>
> Basically, we can fix safari without it (I've added your commit because
> you said that you are going to merge it anyway).
> I think, if we don't define own array props, all we need is to always
> define the only one prop, non-enumerable 0 (revert 7565600
> <https://github.com/mobxjs/mobx/pull/396/commits/7565600e293c9a584fc3fd937e45131126f9d747>
> ).
> So do we need making array props enumerable and loose performance?
>
> —
> You are receiving this because you modified the open/close state.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/mobxjs/mobx/pull/396/files/7565600e293c9a584fc3fd937e45131126f9d747#r70035120>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABvGhAt75uE4TL6tp2DL9HZCVW0xmT-Nks5qTfqwgaJpZM4JHiRJ>
> .
>